openedx/features/pakx/common/utils.py

```diff
@@ -1,35 +1,59 @@
+from logging import getLogger
+from django.conf import settings
+from openedx.features.pakx.cms.custom_settings.models import PartnerSpace
+
+log = getLogger(__name__)
+
+
+def get_active_partner_space(request):
+    """
+    get active partner space form given request object
+    """
+
+    return request.session.get("space")
+
 
 def load_space_in_session(request):
     """
     Set partner space in Session, extracts from query params or sets pakx as default space
     """
 
     space_param = request.GET.get("space", None)
-    active_space = request.session.get("space")
+    active_space = get_active_partner_space(request)
     if active_space is None and space_param is None:
-        set_space_in_session(request, "pakx")  # Set Default space
+        set_space_in_session(request, settings.DEFAULT_PUBLIC_PARTNER_SPACE)  # Set Default space
     elif space_param:
-        set_space_in_session(request, space_param)
-
-    print("***********************\nLOADED SPACE\n*****************************")
+        set_space_in_session(request, space_param.lower())
 
 
 def set_space_in_session(request, space):
     """
     sets given space in given request session
     """
 
-    request.session["space"] = space
-    print("***********************\nUPDATED SPACE\n*****************************")
+    space = PartnerSpace.get_partner_space(space)
+    if space is None:
+        raise Exception("No Partner space found, add a space by visiting <lms>/admin/custom_settings/partnerspace/")
```

Comment:
instead of raising an exception, shouldn't we active PakX/ilmX space?

Reply:
This exception will be raised when Space Model for `PakistanX/ilmX` is not found. At least one default model is required.